crmsh/ra.py

```diff
@@ -542,13 +520,35 @@ def unreq_param(p):
                 rc |= utils.get_check_rc()
         return rc
 
-    def get_adv_timeout(self, op, node=None):
-        if node is not None and op == "monitor":
-            name = monitor_name_node(node)
-        else:
-            name = op
+    def get_op_attr_value(self, op, key, role=None, depth=None):
+        """
+        Get operation's attribute value
+        Multi monitors should be distinguished by role or depth
+        """
         try:
-            return self.actions()[name]["timeout"]
+            # actions_dict example:
+            # {'monitor': [
+            #    {'depth': '0', 'timeout': '20s', 'interval': '10s', 'role': 'Promoted'},
+            #    {'depth': '0', 'timeout': '20s', 'interval': '11s', 'role': 'Unpromoted'}
+            #    ],
+            #  'start': {'timeout': '20s'},
+            #  'stop': {'timeout': '20s'}}
+            actions_dict = self.actions()
+            if not actions_dict:
+                return None
+            if op == 'monitor':
+                if role is None and depth is None:
+                    return actions_dict[op][0][key]
+                if role:
+                    for idx, monitor_item in enumerate(actions_dict[op]):
+                        if monitor_item['role'] == role:
+                            return actions_dict[op][idx][key]
+                if depth:
+                    for idx, monitor_item in enumerate(actions_dict[op]):
+                        if monitor_item['depth'] == depth:
+                            return actions_dict[op][idx][key]
```

Comment:
I added this as a comment in the code
Maybe need some refactors to satisfy more complex cases

Reply:
The discussion was about the naming inconsistency between the advised `depth` and the actually used `OCF_CHECK_LEVEL`, and it would be troublesome to either adapt for  `depth`  or rename it,  hence that was not decided ...  But anyways, the ability to still advise and use multiple levels of monitors for a same role makes total sense.